content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1433 @@
+
+One type of check we could make is to verify the domain name associated with an XRP Ledger address; this is called [account domain verification](xrp-ledger-toml.html#account-verification). When an account's domain is verified, we can could show it like this:
+
+![Screenshot: Step 8, use domain verification](img/javascript-wallet-8.png)
+
+1. In the `library` folder, add a new file `4_helpers.js`. Then add the following contents to that file:
+
+{{ include_code("_code-samples/build-a-wallet/desktop-js/library/8_helpers.js", language="js") }}
+
+The code in the helper class basically issues an [`account_info`](account_info.html) request to look up the account in the ledger.
+
+If the account does exist, the code checks for the [`lsfDisallowXRP` flag](accountroot.html#accountroot-flags). Note that this is an `lsf` (ledger state flag) value because this is an object from the ledger state data; these are different than the flag values the [AccountSet transaction][] uses to configure the same settings.
+
+2. Import the new helper function in`index.js`:
+
+```javascript
+const { initialize, subscribe, saveSaltedSeed, loadSaltedSeed } = require('./library/5_helpers')
+const { sendXrp } = require('./library/7_helpers')
+// Step 8 code additions - start
+const { verify } = require('./library/8_helpers')
+// Step 8 code additions - end
+```
+
+3. At the end of the callback function `ipcMain.on('send-xrp-action', callback)` add the following event handler:
+
+```javascript
+ipcMain.on('send-xrp-action', (event, paymentData) => {
+  sendXrp(paymentData, client, wallet).then((result) => {
+    appWindow.webContents.send('send-xrp-transaction-finish', result)
+  })
+})
+
+// Step 8 code additions - start
+ipcMain.on('destination-account-change', (event, destinationAccount) => {
+  verify(destinationAccount, client).then((result) => {
+    appWindow.webContents.send('update-domain-verification-data', result)
+  })
+})
+// Step 8 code additions - end
+```
+
+3. Modify `view/preload.js` and add the following two functions to `'electronAPI'`:
+
+```javascript
+onDestinationAccountChange: (callback) => {
+  ipcRenderer.send('destination-account-change', callback)
+},
+onUpdateDomainVerificationData: (callback) => {
+  ipcRenderer.on('update-domain-verification-data', callback)
+}
+```
+
+Finally, the code decodes the account's `Domain` field, if present, and performs domain verification using the method imported above.
+
+4. Update the view logic - in `view/template.html` add the following lines just before the `<input>` element with `id="input-destination-address`:
```

Comment:
The new code fully replaces the old one it seems
```suggestion
4. Replace `<input type="text" class="form-control" placeholder="r9jEyy3nrB8D7uRc5w2k3tizKQ1q8cpeHU" id="input-destination-address">` with the following code:
```

Reply:
I definitely messed up my copy + paste here somehow because I ended up with two "To Address" fields... Ended up just resetting to the provided 'answer' for the `template.html` file to keep testing other things.

Reply:
I redid that section, the inline code comments were very skewed.

Reply:
@JST5000 I leave that open for now in case you need to recover that code

Reply:
```html
<div class="input-group mb-3">
    <!-- Step 8 code additions - start -->
    <div class="accountVerificationIndicator">
        <span>Verification status:</span>
    </div>
    <!-- Step 8 code additions - end -->
    <input type="text" class="form-control" placeholder="rP4zcp52pa7ZjhjtU9LrnFcitBUadNW8Xz" id="input-destination-address">
    <span class="input-group-text">To (Address)</span>
</div>
```

Reply:
Ok, should be done by now